src/main/java/com/hermesworld/ais/galapagos/topics/controller/TopicController.java

```diff
@@ -96,6 +99,49 @@ public TopicController(ValidatingTopicService topicService, KafkaClusters kafkaE
         }
     }
 
+    @GetMapping(value = "/api/applicationsmetadata/{envId}/{topicName}", produces = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
I completely vote against this endpoint. This logic (as I understand it: Just used to show only not-yet-added applications in the UI) should be in the UI. You have all information you need there: The current list of producers of the topic, the list of applications available to the user. Just calculate the diff there.